[GTK][WebRTC] webrtc/video-h264.html is failing

philn opened the pull request:
#### 906621c54f03203bc747337e9af1b08f17f96e05
<pre>
[GTK][WebRTC] webrtc/video-h264.html is failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=201267">https://bugs.webkit.org/show_bug.cgi?id=201267</a>

Reviewed by Carlos Garcia Campos.

Advertise constrained-baseline and high H.264 profiles in the GstWebRTC capabilities.

Drive-by: Refactor sframe test expectations in a single section.

* LayoutTests/platform/glib/TestExpectations:
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerPeerConnectionBackend.cpp:
(WebCore::gstreamerRtpCapatiblities):

Canonical link: <a href="https://commits.webkit.org/252409@main">https://commits.webkit.org/252409@main</a>
</pre>

carlosgcampos commented on LayoutTests/platform/glib/TestExpectations:
```diff
@@ -2040,6 +2033,9 @@ webrtc/audio-sframe.html [ Skip ]
 webrtc/sframe-transform-buffer-source.html [ Skip ]
 webrtc/video-sframe.html [ Skip ]
+webkit.org/b/219825 [ Release ] webrtc/sframe-keys.html [ Skip ]
+webkit.org/b/219825 [ Debug ] webrtc/sframe-keys.html [ Skip ]
```
Why do we need the Release/Debug tags here? could it be just one line without any tag?

philn replied:
Yup

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerPeerConnectionBackend.cpp:
```diff
@@ -80,7 +80,7 @@ static RTCRtpCapabilities gstreamerRtpCapatiblities(const String& kind)
     } else {
         capabilities.headerExtensions.append({ "urn:3gpp:video-orientation"_s });
-        capabilities.codecs.reserveCapacity(4);
+        capabilities.codecs.reserveCapacity(6);
```
Isn't 5 enough?

philn replied:
No, in main it's wrong already, we reserve for 4 but append 5.

philn committed:
[GTK][WebRTC] webrtc/video-h264.html is failing
https://bugs.webkit.org/show_bug.cgi?id=201267

Reviewed by Carlos Garcia Campos.

Advertise constrained-baseline and high H.264 profiles in the GstWebRTC capabilities.

Drive-by: Refactor sframe test expectations in a single section.

* LayoutTests/platform/glib/TestExpectations:
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerPeerConnectionBackend.cpp:
(WebCore::gstreamerRtpCapatiblities):

Canonical link: https://commits.webkit.org/252409@main
